core/src/main/scala/org/apache/spark/sql/execution/CompactExecRowToMutableRow.scala

```diff
@@ -31,24 +31,16 @@ import org.apache.spark.sql.types.{DataType, Decimal, DecimalType, StructType}
 import org.apache.spark.unsafe.Platform
 import org.apache.spark.unsafe.types.UTF8String
 
-trait CompactExecRowToMutableRow extends ResultWasNull {
+abstract class CompactExecRowToMutableRow extends ResultNullHolder {
```

Comment:
Why this "wasnull" interface required. Did not get this. Someone would toggle it to non null once the mutable row is created from the compactexecrow?